### DISCUSSION TITLE
Trying to use OME in an EC2 instance but I am not able to make it work

### DISCUSSION THREAD

**Discussion by im-rotaru** [OvenMediaLabs/OvenMediaEngine#1571] [Help] [answered]
Hi,

- I created an EC2 instance with Ubuntu 22.04, SSH into it and installed docker.

- I pulled the docker image airensoft/ovenmediaengine:latest

- I tried to run the OME with this command:

```
sudo docker run --name testOME -d -e OME_HOST_IP=<EC2-instance-public-ip> \
-p 8090:80 -p 1935:1935 -p 9999:9999/udp -p 9000:9000 -p 3333:3333 -p 3478:3478 \
-p 10000-10009:10000-10009/udp airensoft/ovenmediaengine:latest
```

- But when I try to access the demo by going to:
`http://<EC2-instance-public-ip>:8090`
I get
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/0de6747b-e9c5-4e00-a427-c43199501c2b)

- This is how the Security group looks like for inbound:
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/cfed9a08-1a4a-4017-b091-f6fff8b1a014)

Can anyone point me in the right direction please?

**Comment by bchah** [comment]
The HTTP port does not lead to a web interface - it is for HLS live streams so you need to include a path to an active HLS manifest, e.g.` http://<EC2-instance-public-ip>:8090/app/yourlivestream/llhls.m3u8` - it will return 404 if the stream is not live, or 200 along with the playlists if it is.

**Reply by im-rotaru** [reply]
I tried and still connection refused from the broswer or with curl
I can stream to the OvenMediaEngine I just cant see it from the browser
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/377374b7-170b-4486-9aa9-d103dce34772)
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/0169d094-2c63-4ad7-b919-85f41bf2cb2c)
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/b505ed64-b5bd-40a8-9bd2-4b60600c4c84)
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/7d0fb91b-2661-4269-8eea-f48c7d5bd4a6)
![image](https://github.com/AirenSoft/OvenMediaEngine/assets/6125809/a7ace8a8-589d-475f-8d80-dcdac3772931)

**Reply by bchah** [reply]
Something we haven't seen yet is your Server.xml - are you running the LLHLS service on port 80? Or is it bound to another port?

If you are running the service on port 80, then maybe the port is not accessible at the OS level. Have you checked Ubuntu-level firewalls like ufw and selinux to ensure they are not interfering?

And as a sanity check, try stopping OME, apt-get install apache2 and see if you can hit the homepage via the same path.

**Reply by im-rotaru** [reply]
I stopped OME installed apache and i can go to the public ip and see the apache default page

**Comment by naanlizard** [comment]
Looks like your SSL cert isn't loading. Either disable that or ensure you
have a certificate available

Also please post text instead of images, it makes debugging easier. And
please post your server.XML configuration if you need more help

**Reply by bchah** [reply]
I don't think they're using SSL for this test - guessing that they haven't changed LLHLS to port 80 so nothing is running there. As you said, the Server.xml will reveal much! 👍

**Reply by im-rotaru** [reply]
@naanlizard 
[Server.txt](https://github.com/AirenSoft/OvenMediaEngine/files/14812555/Server.txt)
here is the server.xml file (renamed it to .txt since i can't upload .xml files)

@bchah true I currently wanted to test it without SSL, I just wanted to try it with the default settings then to continue with the complex configuration and add a certificate.

**Reply by bchah** [reply, marked as answer]
@SleepyHitman it appears your LLHLS service is running on port 3333, not 80 (unless you've declared the associated environment variable). That is likely the issue.

**Comment by naanlizard** [comment]
You might be required to have a cert to enable the TLS port which is what
it recommends for llhls in the server.xml.

Either way, are these env vars set?

{env:OME_LLHLS_STREAM_PORT:3333}</Port>
 <TLSPort>${env:OME_LLHLS_STREAM_TLS_PORT:3334}

Because if not it's defaulting to ports 3333 and 3334. Which, is that
right? I haven't checked the default server.xml in a while but I'd be
surprised if webrtc and hls can be served on the same port

**Reply by im-rotaru** [reply]
when I first ran the docker container I only set the env var OME_HOST_IP as I saw here:
https://airensoft.gitbook.io/ovenmediaengine/quick-start#run-ovenmediaengine
and also added the -p 8090:80 to be able to acces it with a browser as I saw here:
https://airensoft.gitbook.io/ovenmediaengine/quick-start#run-ovenplayer-demo

**Reply by bchah** [reply]
OME_HOST_IP won't change the port bindings - please have a look through your Server.xml for the appropriate variables and try setting them accordingly. OvenPlayer is a separate piece to OME, it is simply a player you can use when building your application.

**Reply by im-rotaru** [reply]
Thank you for your assistance, both of you.

I will continue trying to make it work with the information you provided.

**Comment by naanlizard** [comment]
That second link is oven player, not ome
